python/python/lance/dataset.py

```diff
@@ -4774,6 +4776,19 @@ def write_dataset(
         and can be retrieved using read_transaction().
         If both `commit_message` and `properties` are provided, `commit_message` will
         override any "lance.commit.message" key in `properties`.
+    data_path_uris: list of str, optional
```

Comment:
Can we use the term "data_file_bases" and "initial_data_file_bases"? I think in general we want to use the "base" terminology to be consistent, and "initial" makes it clear that the field is dedicated to create mode. 

Also we should define the input types a bit better. I think we should accept a list of string for the "data_file_bases" so that it means the list of base names. And we should accept a list of BasePath objects (name + path) for the "initial_data_file_bases" so those get proper names to be referenced later. What do you think?